src/test/java/com/juniorstart/juniorstart/service/ProjectServiceTest.java

```diff
@@ -0,0 +1,228 @@
+    Pageable pageable;
+    List<Project> projectList;
+    List<User> users;
+    UserPrincipal userPrincipal;
+    ProjectRequest projectRequest;
+    Set<Project> projectSet;
+    HashMap<Long, User> creatorMap;
+    Page<Project> page;
+
+    private void initializeUser() throws Exception  {
+        user = new User();
+        user.setPrivateId(UUID.randomUUID());
+        user.setPublicId(21l);
+        user.setEmailVerified(true);
+        user.setName("Some name");
+        user.setProvider(AuthProvider.local);
+        user.setPassword("Somepassword");
+        user.setAge(12);
+        user.setHiddenFromSearch(true);
+        user.setImageUrl("SomeImageUrl");
+        user.setUserStatus(UserStatus.LOOKING_FOR_A_JOB);
+        user.setEmail("some@example.com");
+    }
+
+    private void initializeProject() throws Exception {
+        project = new Project();
+        project.setBody("Some body");
+        project.setDescription("Some description");
+        project.setNumberOfSeats(12l);
+        project.setTitle("Some title");
+        project.setCreatedAt(createdAt);
+        project.setCreatedBy(user.getPublicId());
+        project.setCreatedAt(createdAt);
+        project.setUpdatedBy(user.getPublicId());
+        project.setId(12l);
+        project.setName("Some Easy name");
+    }
+
+    private void initializeProjectRequest() throws Exception {
+        projectRequest = new ProjectRequest();
+        projectRequest.setBody(project.getBody());
+        projectRequest.setDescription(project.getDescription());
+        projectRequest.setName(project.getName());
+        projectRequest.setNumberOfSeats(project.getNumberOfSeats());
+        projectRequest.setRepository(project.getRepository());
+        projectRequest.setTitle(project.getTitle());
+    }
+
+    private void initializeTechnology() throws Exception {
+        technology = Technologies.builder().technologyType(TechnologyType.FRONTEND).description("Some description").id(1l).title("Some title").build();
+        technology.setCreatedAt(createdAt);
+        technology.setUpdatedAt(createdAt);
+        technology.setCreatedBy(user.getPublicId());
+        technology.setUpdatedBy(user.getPublicId());
+    }
+
+    @BeforeEach
+    public void initialize() throws Exception {
+
+        createdAt = new SimpleDateFormat("yyyy-MM-dd").parse("2020-12-31").toInstant();
+        users = new ArrayList<>();
+        initializeUser();
+        initializeProject();
+        projectList = new ArrayList<>();
+        projectList.add(project);
+        page = MockUtil.createMockPage(projectList);
+        userPrincipal = UserPrincipal.create(user);
+        initializeProjectRequest();;
+        initializeTechnology();
+        project.getTechnologies().add(technology);
+        project.setRepository("Some Repo");
+        project.setUpdatedAt(createdAt);
+        projectSet = new HashSet<>();
+        projectSet.add(project);
+        technology.setProjects(projectSet);
```

Comment:
Nieczytelne jak cholera, zastanów się czy każdą metodę musisz ładować na beforeEach. Jak używasz tylko raz w teście to given a nie before each